packages/cli/src/telemetry/index.ts

```diff
@@ -95,6 +110,26 @@ async function checkOptIn(interactive: boolean): Promise<GlobalTelemetryOptions
   return globalOptions;
 }
 
+function getUserEnvironment(): UserEnvironment {
+  return {
+    platform: process.platform,
+    arch: process.arch,
+    nodeVersion: process.version,
+    cliVersion: require('../../package.json').version,
+    upgradesVersion: getDependencyVersion('@openzeppelin/upgrades'),
+    truffleVersion: getDependencyVersion('truffle'),
+    web3Version: getDependencyVersion('web3'),
+  };
+}
+
+function getDependencyVersion(dep: string): string | null {
+  try {
+    return require(`${dep}/package.json`).version;
+  } catch {
+    return null;
```

Comment:
Is this `null` rather than `undefined` for a specific reason? Curious.

Reply:
Not really. I typically use `null` instead of `undefined` for empty values.

Reply:
Actually I think `undefined` is a much better choice. For example using the question mark in an interface member or function argument makes it potentially `undefined`, but not `null`.

Also default function arguments will be set when the argument is `undefined` and not when it's `null`.

Overall I think `undefined` should be used here. What do you think?

Reply:
Good points. Fixed!